subsys/net/lib/aws_jobs/src/aws_jobs.c

```diff
@@ -0,0 +1,381 @@
+/*
+ *copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/mqtt.h>
+#include <logging/log.h>
+#include "aws_jobs.h"
+
+LOG_MODULE_REGISTER(aws_jobs, CONFIG_AWS_JOBS_LOG_LEVEL);
+
+
+/**
+ * @brief Local function used to error check snprintf outputs.
+ *
+ * @param[in] ret Return value of snpritnf.
+ * @param[in] size Max size of what would have been the final string.
+ * @param[in] entry Name of the function it was called from.
+ *
+ */
+static void report_snprintf_err(u32_t ret, u32_t size, char *entry)
```

Comment:
Should this be made "static inline"?

Reply:
Yup will fix.

Reply:
I was hoping to get around it by just returning the error code of the mqtt calls as it would fail when trying to subscribe to malformed topics. However I do see the case where you sort of get an topic that is legal but not really what you wanted.

I'll add error propagation so it will propagate the errors and return the first error found. However this is makes things a bit harder to debug.

Reply:
Assume this is Okei now that we handle the error and propagate them out of the top-level library calls.